### PR TITLE
[bug](if) fix if function not handle const nullable value

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes
in pr https://github.com/apache/doris/pull/22419,
in order to not convert const column so remove those.
Now have test those need get nested  as const value。

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

**Review by BiteTheDDDDt — DISMISSED** [review]

**Comment by zhangstar333** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.91 seconds
 stream load tsv:          512 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17162229221 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.72 seconds
 stream load tsv:          515 seconds loaded 74807831229 Bytes, about 138 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17162474411 Bytes

**Comment by zhangstar333** [comment]
run p0

**Comment by zhangstar333** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.82 seconds
 stream load tsv:          511 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.0 seconds inserted 10000000 Rows, about 344K ops/s
 storage size: 17162186618 Bytes

File: be/src/vec/functions/if.cpp [hunk 1]
```diff
@@ -480,9 +480,8 @@ class FunctionIf : public IFunction {
 
     Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
                         size_t result, size_t input_rows_count) override {
-        const ColumnWithTypeAndName& arg_then = block.get_by_position(arguments[1]);
-        const ColumnWithTypeAndName& arg_else = block.get_by_position(arguments[2]);
-
+        ColumnWithTypeAndName arg_then = block.get_by_position(arguments[1]);
```
Comment (HappenLee) [inline be/src/vec/functions/if.cpp:483]: here maybe a copy ？

File: be/src/vec/functions/if.cpp [hunk 2]
```diff
@@ -494,6 +493,19 @@ class FunctionIf : public IFunction {
...
+        if (auto* then_is_const = check_and_get_column<ColumnConst>(*arg_then.column)) {
+            if (then_is_const->only_null() == false &&
+                check_and_get_column<ColumnNullable>(then_is_const->get_data_column())) {
+                arg_then.column = then_is_const->get_data_column_ptr();
+            }
+        }
+        if (auto* else_is_const = check_and_get_column<ColumnConst>(*arg_else.column)) {
+            if (else_is_const->only_null() == false &&
+                check_and_get_column<ColumnNullable>(else_is_const->get_data_column())) {
+                arg_else.column = else_is_const->get_data_column_ptr();
```
Comment (HappenLee) [inline be/src/vec/functions/if.cpp:505]: the data_column length is 1，may cause bug in block

**Comment by zhangstar333** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.77 seconds
 stream load tsv:          512 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17162192269 Bytes

**Comment by zhangstar333** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.14 seconds
 stream load tsv:          513 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.5 seconds inserted 10000000 Rows, about 338K ops/s
 storage size: 17162331169 Bytes

File: be/src/vec/functions/if.cpp [hunk 3]
```diff
@@ -396,28 +396,64 @@ class FunctionIf : public IFunction {
...
-                     {nullptr, std::make_shared<DataTypeUInt8>(), ""}});
+            // get nullmap from column:
+            // a. nullable column ----> it's a real nullable column get_null_map_column_ptr() / fake nullable, a const nullable
+            // b. not nullable    ----> not have null map, so create const null map
+            Block temporary_block;
+            temporary_block.insert(arg_cond);
+            auto then_nested_null_map =
+                    then_type_is_nullable
+                            ? (then_column_is_const_nullable
+                                       ? DataTypeUInt8().create_column_const_with_default_value(
```
Comment (HappenLee) [inline be/src/vec/functions/if.cpp:437]: why here do not need check the the indeed the null in const column?

Comment (zhangstar333) [reply be/src/vec/functions/if.cpp:437]: the column only null have been handle by `execute_for_null_then_else`

**Comment by zhangstar333** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.11 seconds
 stream load tsv:          512 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17162004949 Bytes

**Comment by zhangstar333** [comment]
run but

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**